### ISSUE TITLE
A Decentralized IPFS Pinning Solution for IPFS Docs

### ISSUE THREAD

**Issue by kbroflovski21** [ipfs/ipfs-docs#771]
## 1. Background

As we have noticed that IPFS Docs is hosted on IPFS (as a matter of course it is IPFS Docs), and learnt that Fleek.co has provided IPFS hosting service for IPFS Docs.

In addition to that, we would like to propose and contribute a fully decentralized IPFS Pinning solution for IPFS Docs, to provide more replicas and further improve availability for IPFS docs, and as well matches the vision of IPFS - decentralization and participation.

## 2. Who are we?

Our solution runs based on IPFS and [Crust Network](https://github.com/crustio). Crust Network is an IPFS-compatible, decentralized storage network, which basically provides IPFS Pinning service in a fully decentralized manner.

Recently [Uniswap](https://github.com/Uniswap/uniswap-interface) has enabled decentralized deployment of its Interface by integrating with Crust Network into its [release workflow](https://github.com/Uniswap/uniswap-interface/blob/main/.github/workflows/release.yaml). The similar integration was later realized by Polkadot. PolkaApps is now be pinned to Crust Network decentralized. The above two examples show that there is a demand for decentralized deployment of websites or dApps in Web 3.0, and Crust is an accurate and workable solution to it.

## 3. Solution Detail

Currently, we have already completed a test case in which we place a storage order in Crust DSM and have IPFS Docs files pinned on IPFS with more than 180 replicas. Check the replicas (CID: QmcxZGzmRqjx5y9j4gzZbrar4g2SiPNYX9Sg3WVfTCfrn1) on [Crust Apps](https://apps.crust.network/#/storage).

These replicas will be pinned by the Crust network for good, for free, albeit as a result of our test case. We want to make it an IPFS Pinning with better availability, security and decentralization level.

## 4. Further solution & proposal

We are constructing a more complete, comprehensive solution, to help websites and DApps including IPFS Docs to enjoy a more decentralized, secure IPFS Pinning/Hosting services.

Solution Components:

* IPFS Gateway. We run a public IPFS gateway (ipfs.decoo.io -to explain, Decoo is a close partner to Crust) which is listed on IPFS’s official [gateway checker](https://ipfs.github.io/public-gateway-checker/).
* Basic tools and technical solutions based on Gateway needed for IPFS Pin & Hosting services, including domain management, comprehensive APIs, etc.
* IPFS Pinning workflow. Such [workflows](https://github.com/crustio/ipfs-crust-pinner) are already realized and adopted by Uniswap, Polkadot Apps, etc. We can also customize workflows for specific applications and scenarios (i.e., IPFS Docs releasing workflow).

## To sum up

We are willing and ready to contribute this decentralized IPFS Pinning and Hosting solution for IPFS Docs, to better help realize its vision of Decentralization and Participation. We suggest that a GitHub workflow could be included in the IPFS Docs releasing workflow, and we can contribute open source code for relevant workflows and solutions.

**Comment by johnnymatthews** [comment]
Anyone is free to host a copy of IPFS Docs, so be all means, go ahead! However, we're not looking to change deployment infrastructure that docs.ipfs.io relies up. I'm gonna close this issue, feel free to expand on your issue if you think I've missed the point.

**Comment by kbroflovski21** [comment]
Hey Johnny, you've made your point very clear. Still we are happy to contribute an alternative pinning solution as we have suggested before. Decoo, a close partner of Crust Network, provides a current solution in which we keep monitoring updates from IPFS Docs and pin the latest version using IPFS/Crust. We can have more than 170 replicas on IPFS as a result of our pinning action. We think this will help to make IPFS Docs better accessed by users across the globe.

To check the pinning result, type in the [CID](https://ipfs.decoo.io/ipfs/QmcZkzeSGv5dPJoCJoXfey9CP5ECDKLq5nLCTEeuVJynBF/) of the latest IPFS Docs version in [Crust's storage explorer](https://apps.crust.network/#/storage) (which shows how many replicas there are as a result of pinning). 

There be a need for an enhanced release workflow for IPFS Docs in the future, please feel free to contact [Decoo.io](https://decoo.io/) or [Crust Network](http://crust.network/), we'll always be happy to contribute!
